src/test/java/com/epam/tc/hw7/JdiTest.java

```diff
@@ -0,0 +1,85 @@
+package com.epam.tc.hw7;
+
+import static com.epam.jdi.light.elements.init.UIFactory.$;
+
+import com.epam.jdi.light.driver.WebDriverUtils;
+import com.epam.jdi.light.elements.init.PageFactory;
+import com.epam.tc.hw7.pages.JdiSite;
+import com.epam.tc.hw7.pages.MetalsPage;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class JdiTest {
+
+    @BeforeSuite(alwaysRun = true)
+    public void beforeSuite() {
+        PageFactory.initSite(JdiSite.class);
+    }
+
+    @AfterSuite(alwaysRun = true)
+    public void afterSuite() {
+        WebDriverUtils.killAllSeleniumDrivers();
+    }
+
+    @Test(dataProvider = "jsonData")
+    public void loginSimpleTest(Map<String, JsonObject> data) {
+        JdiSite.homePage.open();
+        $("img#user-icon").click();
+        $("#name").sendKeys("Roman");
+        $("#password").sendKeys("Jdi1234");
+        $("#login-button").click();
+        $("#user-name").is().displayed();
```

Comment:
Please create separate classes for the objects and forms you use. Hide the logic of working with fields inside. Use page object pattern

Reply:
I found this code in the official JDI docs - why I can't use it?
Also - I tried to use page objects and faced with error - Can't init 'element' 'имялюбогоэлемента' on 'HomePage' - because  system can't create any element except UIElement.
Maybe you know how to fix it?

Reply:
This is a simple example of using JDI without creating additional infrastructure. But this is a bad option in terms of design patterns and the actual use of code in projects. 

Can you show your example where you tried to do this and saw such a problem? Because now the situation is not very clear. If it's more convenient, you can write to me in Teams